src/transaction/_transaction.py

```diff
@@ -541,6 +546,7 @@ def abort(self):
 
             try:
                 self._synchronizers.map(lambda s: s.beforeCompletion(self))
+                self.status = Status.ABORTING
```

Comment:
Probably move that down to line 555, right before the `for rm in self._resources` loop which iterates and aborts.